lib/ide/src/view/text_editor.rs

```diff
@@ -45,54 +88,65 @@ impl TextEditor {
         let text_size    = 16.0;
         let properties   = TextFieldProperties {font,text_size,base_color,size};
         let text_field   = TextField::new(&world,properties);
+        let key_listener = None;
+        let controller_clone = controller.clone_ref();
+        let text_field_clone = text_field.clone();
+        executor::global::spawn(async move {
+            if let Ok(content) = controller_clone.read_content().await {
+                text_field_clone.write(&content);
```

Comment:
Also, I have mixed feelings about design where we open text editor first with no contents, then allow user writing contents, then we overwrite the contents with whatever comes from file manager (remember that e.g. due to some connectivity hiccup the read file answer may come with significant delay).

I'd consider having having the whole text editor creation being async operation depending on reading file contents (so we can editor only after file has been read and we know we can safely edit it).
This is not request for change now (unless you think it is a good idea and easy to implement), just a thought for future work.